dhis-2/dhis-web-api/src/main/resources/openapi/EventsExportController.md

```diff
@@ -131,3 +131,71 @@ Supported properties are `assignedUser`, `assignedUserDisplayName`, `attributeOp
 `completedBy`, `createdAt`, `createdBy`, `deleted`, `enrolledAt`, `enrollment`, `enrollmentStatus`, `event`, `followup`,
 `occurredAt`, `orgUnit`, `orgUnitName`, `program`, `programStage`, `scheduleAt`, `status`, `storedBy`, `trackedEntity`,
 `updatedAt`, `updatedBy`.
+
+### `*.parameter.EventRequestParams.fields`
+
+Get only the specified fields in the JSON response. This query parameter allows you to remove unnecessary fields from
+the JSON response and in some cases decrease the response time. Refer to
+https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/metadata.html#webapi_metadata_field_filter
+for how to use it.
+
+NOTE: this query parameter has no effect on a CSV response!
+
+### `*.parameter.EventRequestParams.filter`
+
+`<filter1>[,<filter2>...]`
+
+Get events matching given filters on data values. A filter is a colon separated data element UID with optional
+operator and value pairs. Example: `filter=H9IlTX2X6SL:sw:A` with operator starts with `sw` followed by a value. Special
+characters like `+` need to be percent-encoded so `%2B` instead of `+`. Multiple operator/value pairs for the same
+data element like `filter=AuPLng5hLbE:gt:438901703:lt:448901704` are allowed. Repeating the same data element UID is not
+allowed. Operator and values are case-insensitive. A user needs access to the attribute to filter on it. A user needs
```

Comment:
Filter are on data values, right? I am not sure I understand why you are talking about attribute here. Did you mean data element or am I missing something?

Reply:
This should be data value or data element. Do you know which one the user needs to have access to and what access?

Reply:
@Bekkalizer @enricocolasante Do you know if the user needs to have access to data values and/or data elements and what access?

Reply:
@Bekkalizer @enricocolasante do you know if the user needs to have access to data values and/or data elements and what access? I would love to get this merged and this is the only missing piece 😄

Reply:
since you are on export, user needs to have `Permission.DATA_READ` access at program stage level. We don't have data sharing at data element level.

Reply:
@abyot and what access is needed to filter on attributes?

Reply:
like data elements, we don't have data sharing at attribute level. We instead rely on data sharing at program level. so you just check data read access at program level.

by the way for both data element and attribute we have metadata sharing and a user needs to have metadata read access to get data read access.

Reply:
@abyot please let me know if the changed sentence is correct 😄